packages/playwright-test/src/runner/uiMode.ts

```diff
@@ -81,49 +81,58 @@ class UIMode {
     return status;
   }
 
-  async showUI() {
-    this._page = await openTraceViewerApp([], 'chromium', { app: 'uiMode.html', headless: isUnderTest() && process.env.PWTEST_HEADED_FOR_TEST !== '1' });
+  async showUI(openInBrowser: boolean) {
+    const exitPromise = new ManualPromise();
+    let queue = Promise.resolve();
+
+    this._transport = {
+      dispatch: async (method, params) => {
+        if (method === 'exit') {
+          exitPromise.resolve();
+          return;
+        }
+        if (method === 'watch') {
+          this._watchFiles(params.fileNames);
+          return;
+        }
+        if (method === 'open' && params.location) {
+          open('vscode://file/' + params.location).catch(e => this._originalStderrWrite.call(process.stderr, String(e)));
+          return;
+        }
+        if (method === 'resizeTerminal') {
+          process.stdout.columns = params.cols;
+          process.stdout.rows = params.rows;
+          process.stderr.columns = params.cols;
+          process.stderr.columns = params.rows;
+          return;
+        }
+        if (method === 'stop') {
+          void this._stopTests();
+          return;
+        }
+        queue = queue.then(() => this._queueListOrRun(method, params));
+        await queue;
+      },
+
+      onclose: () => exitPromise.resolve(),
+    };
+    await showTraceViewer([], 'chromium', {
+      app: 'uiMode.html',
+      headless: isUnderTest() && process.env.PWTEST_HEADED_FOR_TEST !== '1',
+      transport: this._transport,
```

Comment:
we should directly have `--ui-web-host` here since Codespaces requires to have `0.0.0.0` as a listening one.